tests/ttsim/tt_dag_elements/test_vectorization.py

```diff
@@ -681,3 +681,54 @@ def test_already_vectorized_func(xnp):
         already_vectorized_func(xnp.array([-1, 0, 2, 3]), xnp),
         xnp.array([0, 0, 4, 6]),
     )
+
+
+def test_vectorize_function_annotations_numpy():
+    def f(a, x: int, y: float, z: bool, p1: str, p2: dict[str, float]) -> float:  # noqa: ARG001
+        return 1.0
+
+    vectorized = vectorize_function(
+        f,
+        vectorization_strategy="vectorize",
+        backend="numpy",
+        xnp=numpy,
+    )
+
+    expected_annotations = {
+        "a": "xnp.ndarray",
+        "x": "numpy.typing.NDArray[numpy.int64]",
+        "y": "numpy.typing.NDArray[numpy.float64]",
+        "z": "numpy.typing.NDArray[numpy.bool_]",
+        "p1": "str",
+        "p2": "dict[str, float]",
+        "return": "numpy.typing.NDArray[numpy.float64]",
+    }
+    assert inspect.get_annotations(vectorized) == expected_annotations
+
+
+def test_vectorize_function_annotations_jax():
```

Comment:
```suggestion
def test_vectorize_function_annotations(backend, xnp):
```
See `conftest.py`. Will be automatically chosen based on flags. Sorry, this I should really have mentioned since it is undocumented so far...

Run jax tests via `pixi run tests-jax tests` now (the last `tests` just because otherwise the GETTSIM tests will run and mostly fail).